File: Data Structures & Algorithms/missing-number-in-arithmetic-progression/submission-1.java
class Solution {
    public int missingNumber(int[] arr) {
        
        int r = (arr[arr.length-1] - arr[0])/(arr.length);

        

        int i = arr.length-1;

        while(i>0){
            if(arr[i]-r != arr[i-1]){
                return arr[i]-r;
            } else{
                i--;
            }
        }

        return arr[0];
    }
}

File: Data Structures & Algorithms/counting-elements/submission-1.kt
class Solution {
    fun countElements(arr: IntArray): Int {

        var seen = mutableSetOf<Int>()

        arr.forEach{
            seen.add(it)
        }

        var ans = 0

        arr.forEach{
            if((it+1) in seen){
                ans++
            }
        }

        return ans
    }
}
